Src/library/base/elks/kernel/tuple.e

```diff
@@ -369,22 +443,65 @@ feature -- Element change
 			valid_index: valid_index (index)
 			valid_type_for_index: valid_type_for_index (v, index)
 		do
-			inspect eif_item_type ($Current, index)
-			when boolean_code then eif_put_boolean_item_with_object ($Current, index, $v)
-			when character_8_code then eif_put_character_8_item_with_object ($Current, index, $v)
-			when character_32_code then eif_put_character_32_item_with_object ($Current, index, $v)
-			when real_64_code then eif_put_real_64_item_with_object ($Current, index, $v)
-			when real_32_code then eif_put_real_32_item_with_object ($Current, index, $v)
-			when pointer_code then eif_put_pointer_item_with_object ($Current, index, $v)
-			when natural_8_code then eif_put_natural_8_item_with_object ($Current, index, $v)
-			when natural_16_code then eif_put_natural_16_item_with_object ($Current, index, $v)
-			when natural_32_code then eif_put_natural_32_item_with_object ($Current, index, $v)
-			when natural_64_code then eif_put_natural_64_item_with_object ($Current, index, $v)
-			when integer_8_code then eif_put_integer_8_item_with_object ($Current, index, $v)
-			when integer_16_code then eif_put_integer_16_item_with_object ($Current, index, $v)
-			when integer_32_code then eif_put_integer_32_item_with_object ($Current, index, $v)
-			when integer_64_code then eif_put_integer_64_item_with_object ($Current, index, $v)
-			when Reference_code then eif_put_reference_item_with_object ($Current, index, $v)
+			inspect item_code (index)
+			when boolean_code then
+				check valid_type_for_index: attached {BOOLEAN_REF} v as l_b then
```

Comment:
I'm lost here. I thought that one of the purposes of 'check' was to make explicit to the compiler something which is obvious for the programmer. Because the property is already expressed in the precondition, as a programmer we know that it's a boolean. So it's a 'check'. Using 'if' would mean that as a programmer we are not sure that it's a boolean, in which case it would require an 'else' clause to describe what to do when it's not a boolean.